library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableComplexComponentImpl.java

```diff
@@ -0,0 +1,202 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.state.spillable;
+
+import java.util.List;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.apex.malhar.lib.utils.serde.Serde;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.netlet.util.Slice;
+
+/**
+ * Created by tfarkas on 6/12/16.
+ */
+public class SpillableComplexComponentImpl implements SpillableComplexComponent
+{
+  private List<SpillableComponent> componentList = Lists.newArrayList();
+
+  private boolean isRunning = false;
+  private boolean isInWindow = false;
+
+  @NotNull
+  private SpillableStateStore store;
+
+  @NotNull
+  private SpillableIdentifierGenerator identifierGenerator;
+
+  public SpillableComplexComponentImpl(SpillableStateStore store)
```

Comment:
Isn't this missing no-args constructor? Kryo serialization may fail. 
I usually add the following test in each non-transient component to make sure they are kryo serializable:
```
  @Test
  public void testSerde() throws IOException
  {
    ManagedStateImpl deserialized = KryoCloneUtils.cloneObject(testMeta.managedState);
    Assert.assertEquals("num buckets", deserialized.getNumBuckets(), testMeta.managedState.getNumBuckets());
  }
```